src/helpers/__tests__/global-overrides.test.js

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright (c) 2018-2020 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+
+// @flow
+
+import * as React from 'react';
+import {
+  fireEvent,
+  getByTestId,
+  getByText,
+  render,
+} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect.js';
+import {withOverrides} from '../overrides.js';
+import type {OverrideObjectT} from '../overrides.js';
+import ThemeProvider from '../../styles/theme-provider.js';
+import BaseProvider from '../base-provider.js';
+
+type withOverridesT = {
+  overrides: OverrideObjectT,
+  forwardedRef?: React.Ref<any>,
+};
+
+describe('Global Overrides', () => {
+  test('passes global overrides through ThemeProvider', () => {
+    // Mock behavior of baseui component
+    const TestElement = (props: withOverridesT) => {
+      return (
+        <div {...props.overrides.props} style={props.overrides.style}></div>
+      );
+    };
+    const TestComponent = withOverrides(TestElement, 'TestComponent');
+    const {container} = render(
+      // $FlowFixMe
+      <ThemeProvider
+        overrides={{
+          TestComponent: {
+            props: {
+              'data-testid': 'global-overrides',
+            },
+            style: {
+              color: 'red',
+            },
+          },
+        }}
+      >
+        {/* $FlowFixMe */}
+        <TestComponent />
+      </ThemeProvider>,
+    );
+    const testComponent = getByTestId(container, 'global-overrides');
+    expect(testComponent).toBeTruthy();
+    expect(testComponent).toHaveStyle('color: red');
+  });
+  test('passes global overrides through BaseProvider', () => {
```

Comment:
I think this needs a test that uses an actual baseui component instead of just components defined inline here, like setting the `borderWidth` for an `Input` component.